### PR TITLE
Switch A2 to semanticz-connectors #4

### PR DESCRIPTION
**Pull request by alexey-lukashov**
- added hook to support custom facets
- protected methods to override json document creation when needed
- switched RdfStoreService from A2 to semanticz connectors
- copied ResultSetMapper https://github.com/cognizone/semanticz-elastic-indexer/issues/8

### CODE REVIEW

File: semanticz-elastic-indexer-orchestrator/src/main/java/zone/cogni/semanticz/indexer/orchestrator/IndexOrchestrator.java [hunk 1]
```diff
@@ -11,9 +11,9 @@
 import org.slf4j.LoggerFactory;
 import org.springframework.http.ResponseEntity;
 import org.thymeleaf.TemplateEngine;
-import zone.cogni.asquare.rdf.ResultSetMapper;
-import zone.cogni.asquare.triplestore.RdfStoreService;
-import zone.cogni.sem.jena.model.ResultSetDto;
+import zone.cogni.semanticz.connectors.general.RdfStoreService;
```
Comment (alexey-lukashov) [inline semanticz-elastic-indexer-orchestrator/src/main/java/zone/cogni/semanticz/indexer/orchestrator/IndexOrchestrator.java:14]: https://github.com/cognizone/semanticz-elastic-indexer/issues/8

File: semanticz-elastic-indexer-orchestrator/src/main/java/zone/cogni/semanticz/indexer/tools/QuerySolutionDto.java
Comment (natan-cox-cognizone) [inline semanticz-elastic-indexer-orchestrator/src/main/java/zone/cogni/semanticz/indexer/tools/QuerySolutionDto.java:1]: what is QuerySolutionDto trying to do?

File: semanticz-elastic-indexer-orchestrator/src/main/java/zone/cogni/semanticz/indexer/tools/RdfBooleanNodeDto.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package zone.cogni.semanticz.indexer.tools;
+
+
+public class RdfBooleanNodeDto extends RdfNodeDto<Boolean> {
+
+    public static RdfBooleanNodeDto create(Boolean value) {
```
Comment (natan-cox-cognizone) [inline semanticz-elastic-indexer-orchestrator/src/main/java/zone/cogni/semanticz/indexer/tools/RdfBooleanNodeDto.java:6]: RdfBooleanNodeDto extends RdfNodeDto which has like 5 fields but create method only sets 1 of those 5 fields?

File: semanticz-elastic-indexer-orchestrator/src/main/java/zone/cogni/semanticz/indexer/tools/RdfNodeDto.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
+package zone.cogni.semanticz.indexer.tools;
+
+
+public class RdfNodeDto<T> {
+
+    private String name;
+    private String type;
```
Comment (natan-cox-cognizone) [inline semanticz-elastic-indexer-orchestrator/src/main/java/zone/cogni/semanticz/indexer/tools/RdfNodeDto.java:7]: what is type? you already have datatype and language?

File: semanticz-elastic-indexer-orchestrator/src/main/java/zone/cogni/semanticz/indexer/tools/RdfNodeDto.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
+package zone.cogni.semanticz.indexer.tools;
+
+
+public class RdfNodeDto<T> {
```
Comment (natan-cox-cognizone) [inline semanticz-elastic-indexer-orchestrator/src/main/java/zone/cogni/semanticz/indexer/tools/RdfNodeDto.java:4]: not sure... but this feels like a wrapper for RDFNode?!

File: semanticz-elastic-indexer-orchestrator/src/main/java/zone/cogni/semanticz/indexer/tools/RdfStringNodeDto.java [hunk 5]
```diff
@@ -0,0 +1,18 @@
+package zone.cogni.semanticz.indexer.tools;
+
+public class RdfStringNodeDto extends RdfNodeDto<String> {
+
+    public static RdfStringNodeDto create(String value) {
```
Comment (natan-cox-cognizone) [inline semanticz-elastic-indexer-orchestrator/src/main/java/zone/cogni/semanticz/indexer/tools/RdfStringNodeDto.java:5]: I do not understand these constructors... they only set part of the fields.

File: semanticz-elastic-indexer-orchestrator/src/main/java/zone/cogni/semanticz/indexer/tools/RdfStringNodeDto.java [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+        RdfStringNodeDto node = new RdfStringNodeDto();
+        node.setValue(value);
+        return node;
+    }
+
+    public static RdfStringNodeDto create(String value, String name, String type) {
```
Comment (natan-cox-cognizone) [inline semanticz-elastic-indexer-orchestrator/src/main/java/zone/cogni/semanticz/indexer/tools/RdfStringNodeDto.java:11]: I do not understand these constructors... they only set part of the fields.

File: semanticz-elastic-indexer-orchestrator/src/main/java/zone/cogni/semanticz/indexer/tools/ResultSetDto.java [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+
+
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class ResultSetDto {
```
Comment (natan-cox-cognizone) [inline semanticz-elastic-indexer-orchestrator/src/main/java/zone/cogni/semanticz/indexer/tools/ResultSetDto.java:11]: wrapping Jena resultset?

File: semanticz-elastic-indexer-orchestrator/src/main/java/zone/cogni/semanticz/indexer/tools/RdfNodeDto.java (hunk 4 above)
Comment (alexey-lukashov) [reply semanticz-elastic-indexer-orchestrator/src/main/java/zone/cogni/semanticz/indexer/tools/RdfNodeDto.java:4]: exactly

File: semanticz-elastic-indexer-orchestrator/src/main/java/zone/cogni/semanticz/indexer/tools/ResultSetDto.java (hunk 7 above)
Comment (alexey-lukashov) [reply semanticz-elastic-indexer-orchestrator/src/main/java/zone/cogni/semanticz/indexer/tools/ResultSetDto.java:11]: yes

File: semanticz-elastic-indexer-orchestrator/src/main/java/zone/cogni/semanticz/indexer/tools/RdfNodeDto.java (hunk 3 above)
Comment (alexey-lukashov) [reply semanticz-elastic-indexer-orchestrator/src/main/java/zone/cogni/semanticz/indexer/tools/RdfNodeDto.java:7]: it seems that type is always "literal" here

File: semanticz-elastic-indexer-orchestrator/src/main/java/zone/cogni/semanticz/indexer/tools/QuerySolutionDto.java
Comment (alexey-lukashov) [reply semanticz-elastic-indexer-orchestrator/src/main/java/zone/cogni/semanticz/indexer/tools/QuerySolutionDto.java:1]: just wrapper for jena QuerySolution

File: semanticz-elastic-indexer-orchestrator/src/main/java/zone/cogni/semanticz/indexer/tools/RdfStringNodeDto.java (hunk 5 above)
Comment (alexey-lukashov) [reply semanticz-elastic-indexer-orchestrator/src/main/java/zone/cogni/semanticz/indexer/tools/RdfStringNodeDto.java:5]: confusing indeed, as I understand its mostly about setting value

File: semanticz-elastic-indexer-orchestrator/src/main/java/zone/cogni/semanticz/indexer/tools/ResultSetMapper.java [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+    private ResultSetMapper() {
+    }
+
+    public static ResultSetDto resultSetToResultSetDto(ResultSet resultSet) {
+        return resultSetToResultSetDto(resultSet, null);
+    }
+
+    public static Integer resultSetToAmount(ResultSet resultSet) {
+        ResultSetDto resultSetDto = resultSetToResultSetDto(resultSet, null);
+        String strAmount = resultSetDto.collectPropertyValue("amount");
```
Comment (natan-cox-cognizone) [inline semanticz-elastic-indexer-orchestrator/src/main/java/zone/cogni/semanticz/indexer/tools/ResultSetMapper.java:24]: special case amount??

**Pull request merged**